ftva_lab_data/views.py

```diff
@@ -524,3 +524,51 @@ def get_external_search_results(
             "Invalid search type specified.",
             status=400,
         )
+
+
+def generate_metadata_json(
+    request: HttpRequest, record_id: int, inventory_number: str
+) -> JsonResponse | str:
+    """Generate a  JSON metadata record for a given inventory number,
+    by combining data from Alma, Filemaker, and Django.
+
+    :param request: The HTTP request object.
+    :param record_id: The ID of the Django record to use.
+    :param inventory_number: The inventory number to search for in Alma and Filemaker.
+    :return: A JSON record containing the combined metadata.
+    """
+
+    # Get Django record
+    django_record = SheetImport.objects.get(pk=record_id)
+    # Transform the Django record into a dict
+    django_record_data = {
+        field.name: getattr(django_record, field.name)
+        for field in django_record._meta.fields
+    }
+
+    # Get Alma records
+    sru_client = AlmaSRUClient()
+    bib_records = sru_client.search_by_call_number(inventory_number)
+    bib_records_count = len(bib_records)
+
+    # Get Filemaker records
+    user = settings.FILEMAKER_USER
+    password = settings.FILEMAKER_PASSWORD
+    fm_client = FilemakerClient(user=user, password=password)
+    fm_records = fm_client.search_by_inventory_number(inventory_number)
+    fm_records_count = len(fm_records)
+
+    # If Alma and FM records are unique, generate JSON metadata
+    if bib_records_count == 1 and fm_records_count == 1:
+        metadata = get_mams_metadata(bib_records[0], fm_records[0], django_record_data)
+        # TODO: render a template with the metadata. Returning JSON for now.
+        return JsonResponse(metadata)
+
+    # Otherwise, Alma and/or FM records are either not unique or not found,
+    # so return a message with the counts of records found
+    message = (
+        f"Metadata not generated because the search for inventory number {inventory_number} "
+        f"found {bib_records_count} Alma and {fm_records_count} Filemaker records."
+    )
+    # TODO: render a template with the message. Returning JSON for now.
+    return JsonResponse({"message": message}, status=400)
```

Comment:
Since this view is not intended to be called from outside of Django, the `status` can be removed - which it probably would be when switching to a template anyhow.